doc/rst/language/evolution.rst

```diff
@@ -51,6 +54,96 @@ Additionally, several ``c_string`` methods are deprecated without replacement:
 An equivalent for ``.size`` is the unstable procedure ``strLen(x)`` in the
 ``CTypes`` module.
 
+.. _readme-evolution.ref-if-modified-deprecation:
+
+The default intent for arrays and records
+*****************************************
+
+In version 1.32, arrays and records now always have a default intent of
+``const``. Previously, these types would be passed by ``const ref`` intent,
+unless they were modified and then they were passed by ``ref`` intent. This
+change was motivated by improving the consistency across types and making
+potential problems more apparent.
```

Comment:
I would probably append this to the end of the first sentence: "...of const, meaning that if arrays and records are modified inside of a loop body, they must be specified as `ref` intent via a `with` clause." I see you mention this later, but I think that putting it right at the beginning would be helpful.

Also, I think it would be more clear if we said something like "the compiler would treat these types as `const ref` or `ref`", rather than just "these types were passed".

Reply:
Rewritten to the following paragraph

```
In version 1.32, arrays and records now always have a default intent of
``const``. This means that if arrays and records are modified inside of a loop
body, a function, or a task body they must use a ``ref`` intent. This also
means that record methods which modify their implicit ``this`` argument must
also use a ``ref`` intent. Previously, the compiler would treat these types as
either ``const ref`` intent or ``ref`` intent, depending on if they were
modified. This change was motivated by improving the consistency across types
and making potential problems more apparent.
```